src/components/modules/newsletter/index.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  StyledNewsletterContentWrapper,
+  StyledNewsletterFormContentWrapper,
+  StyledNewsletterFormWrapper,
+  StyledNewsletterGradientWrapper,
+  StyledNewsletterTitle,
+  StyledNewsletterWrapper,
+} from './newsletter.styles';
+import type { NewsletterProps } from './newsletter.types';
+import { Input } from '../input';
+
+import { Button } from '@/components/elements/button';
+import { Gradient } from '@/components/elements/gradient';
+import { Text } from '@/components/elements/text';
+import type { TextProps } from '@/components/elements/text/text.types';
+
+export function Newsletter({
+  $variant = 'default',
+  title,
+  text,
+  supportText,
+  buttonText = 'Sign up',
+  action,
```

Comment:
Could we make this optional and set the action default as default  😄. So, there won't be a need to worry about setting an action if the behavior is the default endpoint.

```suggestion
  action = 'default',
```